### PR TITLE
7 Navigation Bar

### LINKED ISSUE #7: Navigation Bar

### PR DESCRIPTION
**Pull request by Cooljawty**
# Description

Provides a widget for navigating to different pages. The items on the navbar is defined by each page.

Fixes #7 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x]  Test navigation between pages. In this case two diffrent test pages are created each with a NavBar linking the two.

**Test Configuration**:

* Firmware version: Android 14
* Hardware: Google Pixel 3a (emulated)
* Toolchain: Android API 32
* SDK: sdk_gphone6_65

# Demonstration
For this demonstration the dashboard page was edited to include a NavBar with links to relevent pages
navbar

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by S1robe** [comment]
Please include a short video or pictures of what your changed version looks like, I am not sure what this will look like.

Thank you,

**Comment by Cooljawty** [comment]
> Please include a short video or pictures of what your changed version looks like, I am not sure what this will look like.
> 
> Thank you,

I attached a video in the demonstration section. Guess it's now showing a preview

**Comment by S1robe** [comment]
I see, thank you. Sorry, my mistake. This looks really good!

Can you see about making it persist between pages, where the buttons on the navbar change with the page that it is being displayed over?

Please let me know if you need anything else!

**Review by S1robe — CHANGES_REQUESTED** [review]
Thank you for your contributions.
Can you make it persist across the different pages that we already have?

**Comment by caelanot** [comment]
Enum/Boolean for push type (only some pages should have the back button go to previous page, settings for example)

File: app/lib/provider/theme_settings.dart [hunk 1]
```diff
@@ -27,8 +30,10 @@ class ThemeSettings with ChangeNotifier {
           fontSize: 14,
           fontWeight: FontWeight.bold),
     ),
-    primaryColor: Colors.deepPurple
-
+    primaryColor: Colors.deepPurple,
+			navigationBarTheme: const NavigationBarThemeData(
+				labelBehavior: NavigationDestinationLabelBehavior.alwaysHide,
+			),
```
Comment (Benek6h) [inline app/lib/provider/theme_settings.dart:36]: Why would this need to be in the app's theme settings? They don't change across the light and dark themes, so this shouldn't be in the theme settings. It should be in the nav bars initial build.

File: app/lib/uiwidgets/navbar.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	final List<Destination> destinations;
+
+	final int selectedIndex;
+
+	const NavBar({Key? key, required this.destinations, this.selectedIndex = 0}) : super(key: key);
+
+	@override
+	Widget build(BuildContext context) {
+		return NavigationBar(
+			destinations: destinations.map((destination) => destination.toWidget()).toList(),
```
Comment (Benek6h) [inline app/lib/uiwidgets/navbar.dart:20]: Navbar theme should be here

**Review by Benek6h — CHANGES_REQUESTED** [review]
Please look over my comments. Good job with the test cases!

**Review by Benek6h — COMMENTED** [review]
Please refer to my comment about your navbar_test.dart file

File: app/integration_test/navbar_test.dart [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+						destination: TestPage.route(),
+					),
+					Destination(
+						label: "Page 2",
+						icon: Icons.turn_right,
+						destination: TestPage2.route(),
+					),
+				],
+			),
+		);
```
Comment (Benek6h) [inline app/integration_test/navbar_test.dart:81]: This is good, but this doesn't test the navbar in the app environment.
The skeleton test that you had before was good. It iterated through all the pages that it should (dashboard, entries, settings, etc.).
To run it in the app environment, you need to import main as app and run it as app.main()

**Review by S1robe — CHANGES_REQUESTED** [review]
Great commits, but integration tests were not run, please run 'flutter test integration_test --branch' locally, which will create a different lcov.info file and then re commit so we can get an updated coverage metric.

Thank you!

**Review by caelanot — APPROVED** [review]
Thanks for adding the switch state to the push type for the navigation for different pages.

**Review by S1robe — APPROVED** [review]
Excellent PR, thank you for the contribution. Lets discuss sometime about adding custom animations to these transitions. 

Maybe if we did like a page turn or page flip animation like a book or notepad?

Thanks.

**Review by S1robe — APPROVED** [review]
Changes were made that resolve these requests. Resolving on Katrina's behalf.

**Review by Benek6h — APPROVED** [review]
Thank you for implementing my requested changes! Great job, looks amazing!

**Pull request merged**